web/src/main/java/com/epam/esm/controllers/UserController.java

```diff
@@ -0,0 +1,114 @@
+package com.epam.esm.controllers;
+
+import com.epam.esm.dto.OrderDto;
+import com.epam.esm.entities.Order;
+import com.epam.esm.entities.Tag;
+import com.epam.esm.entities.User;
+import com.epam.esm.exception.ResourceNotFoundException;
+import com.epam.esm.service.OrderService;
+import com.epam.esm.service.TagService;
+import com.epam.esm.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.constraints.Min;
+import java.util.List;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+/**
+ *  The class that represents an API for basic user-related application operations
+ *  @author Yauheni Vezhnavets
+ **/
+
+@RestController
+@RequestMapping ("/users")
+public class UserController {
+
+    private static final String JSON = "application/json";
+    private static final String USER_ID = "userId";
+    private static final String GIFT_CERTIFICATE_ID = "certificateId";
+
+    private final UserService<User> userService;
+    private final OrderService <Order> orderService;
+    private final TagService<Tag> tagService;
+
+    @Autowired
+    public UserController(UserService<User> userService, OrderService<Order> orderService, TagService <Tag> tagService) {
+
+        this.userService = userService;
+        this.orderService = orderService;
+        this.tagService = tagService;
+    }
+
+    /**
+     * Returns an {@link ResponseEntity} object contained {@link HttpStatus} status and a {@link List}
+     * list of {@link User} users retrieved from database.
+     *
+     * @return {@link ResponseEntity} contained both {@link HttpStatus} status and {@link List} of {@link User} users.
+     */
+    @GetMapping(produces = JSON, params = {"page"})
+    public ResponseEntity<List<User>> getUsers( @RequestParam(defaultValue = "1") @Min(1) int page) {
+
+        List<User> listOfUsers = userService.findUsers(page);
+        listOfUsers.stream().forEach(
+                user -> user.add(
+                        linkTo(methodOn(UserController.class).getUsers(page)).withRel("getUsers"),
```

Comment:
Improvement: yo can use Resource assembler to populate hateoas links https://www.logicbig.com/tutorials/spring-framework/spring-hateoas/resource-assembler.html